lib/web_ui/test/window_test.dart

```diff
@@ -301,6 +301,30 @@ void testMain() {
     expect(actualState['state4']['substate2'], 'string2');
   });
 
+  test('routeInformationUpdated can handle uri',
+      () async {
+    await window.debugInitializeHistory(TestUrlStrategy.fromEntry(
+      const TestHistoryEntry('initial state', null, '/initial'),
+    ), useSingle: false);
+    expect(window.browserHistory, isA<MultiEntriesBrowserHistory>());
+
+    // routeInformationUpdated does not
+    final Completer<void> callback = Completer<void>();
+    window.sendPlatformMessage(
+      'flutter/navigation',
+      const JSONMethodCodec().encodeMethodCall(const MethodCall(
+        'routeInformationUpdated',
+        <String, dynamic>{
+          'uri': 'http://myhostname.com/baz',
```

Comment:
Maybe make this uri a little more complete to make sure we are capturing query params and fragment?